src/frontend/js/components/UserLogin/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { Spinner } from 'components/Spinner';
+import { User } from 'types/User';
+import { handle } from 'utils/errors/handle';
+import { Maybe, Nullable } from 'utils/types';
+import { useAsyncEffect } from 'utils/useAsyncEffect';
+
+const messages = defineMessages({
+  logIn: {
+    defaultMessage: 'Log in',
+    description: 'Text for the login button.',
+    id: 'components.UserLogin.logIn',
+  },
+  logOut: {
+    defaultMessage: 'Log out',
+    description: 'Text for the logout button.',
+    id: 'components.UserLogin.logOut',
+  },
+  signUp: {
+    defaultMessage: 'Sign up',
+    description: 'Text for the signup button.',
+    id: 'components.UserLogin.signup',
+  },
+  spinnerText: {
+    defaultMessage: 'Loading login status...',
+    description: 'Accessibility text for the spinner in the login area.',
+    id: 'components.UserLogin.spinnerText',
+  },
+});
+
+interface UserLoginProps {
+  loginUrl: string;
+  logoutUrl: string;
+  signupUrl: string;
+}
+
+export const UserLogin = ({
+  loginUrl,
+  logoutUrl,
+  signupUrl,
+}: UserLoginProps) => {
+  /**
+   * `user` is:
+   * - `undefined` when we have not made the `whoami` request yet;
+   * - `null` when the user is anonymous or the request failed;
+   * - a user object when the user is logged in.
+   */
+  const [user, setUser] = useState<Maybe<Nullable<User>>>(undefined);
+
+  useAsyncEffect(async () => {
+    try {
+      const response = await fetch('/api/v1.0/users/whoami/', {
```

Comment:
the `/api/v1.0` part of the endoint is not in a constant ? The day we will increment the api version we will have to edit all files ? You should probably open an issue for this ?

Reply:
IMO it's not a big deal given the size of Richie's JS codebase but sure I can open an issue.